pkg/adapterManager/manager2_test.go

```diff
@@ -0,0 +1,326 @@
+		{badHandler: true},
+	} {
+		t.Run(fmt.Sprintf("%#v", s), func(t *testing.T) {
+			fp := &fakeProc{
+				err:         s.callErr,
+				checkResult: s.cr,
+			}
+			tr := newTemplateRepo(tname, fp)
+			var resolveErr error
+			if s.resolveErr {
+				resolveErr = s.callErr
+			}
+			rt := newRuntimeState("myhandler", "i1", s.tn, resolveErr, s.badHandler)
+			m := NewManager2(nil, tr, rt, gp)
+
+			cr, err := m.Check(context.Background(), nil)
+
+			checkError(t, s.callErr, err)
+
+			if s.callErr != nil {
+				return
+			}
+			if fp.called != s.ncalled {
+				t.Fatalf("got %v, want %v", fp.called, s.ncalled)
+			}
+			if s.ncalled == 0 {
+				return
+			}
+			if cr == nil {
+				t.Fatalf("got %v, want %v", cr, fp.checkResult)
+			}
+			if !reflect.DeepEqual(fp.checkResult.Status.Code, cr.Status.Code) {
+				t.Fatalf("got %v, want %v", *cr, fp.checkResult)
+			}
+		})
+	}
+
+	gp.Close()
+}
+
+func TestQuota2(t *testing.T) {
+	gp := pool.NewGoroutinePool(1, true)
+	tname := "metric1"
+	badTname := "metric2"
+	err1 := errors.New("internal error")
+
+	for _, s := range []struct {
+		tn         string
+		callErr    error
+		resolveErr bool
+		ncalled    int
+		badHandler bool
+		cr         adapter.QuotaResult2
+	}{{tn: tname, ncalled: 1},
+		{tn: tname, ncalled: 1, cr: adapter.QuotaResult2{Amount: 200}},
+		{tn: tname, ncalled: 1, cr: adapter.QuotaResult2{Amount: 200, Status: status.WithPermissionDenied("bad user")}},
+		{tn: tname, callErr: err1},
+		{tn: tname, callErr: err1, resolveErr: true},
+		{tn: badTname},
+		{badHandler: true},
+	} {
+		t.Run(fmt.Sprintf("%#v", s), func(t *testing.T) {
+			fp := &fakeProc{
+				err:         s.callErr,
+				quotaResult: s.cr,
+			}
+			tr := newTemplateRepo(tname, fp)
+			var resolveErr error
+			if s.resolveErr {
+				resolveErr = s.callErr
+			}
+			rt := newRuntimeState("myhandler", "i1", s.tn, resolveErr, s.badHandler)
+			m := NewManager2(nil, tr, rt, gp)
+
+			cr, err := m.Quota(context.Background(), nil,
+				&aspect.QuotaMethodArgs{
+					Quota: "i1",
+				})
+
+			checkError(t, s.callErr, err)
+
+			if s.callErr != nil {
+				return
+			}
+			if fp.called != s.ncalled {
+				t.Fatalf("got %v, want %v", fp.called, s.ncalled)
+			}
+			if s.ncalled == 0 {
+				return
+			}
+			if cr == nil {
+				t.Fatalf("got %v, want %v", cr, fp.quotaResult)
+			}
+			if !reflect.DeepEqual(fp.quotaResult.Status.Code, cr.Status.Code) {
+				t.Fatalf("got %v, want %v", *cr, fp.quotaResult)
+			}
+		})
+	}
+
+	gp.Close()
+}
+
+func TestPreprocess2(t *testing.T) {
+	m := Manager2{}
+
+	err := m.Preprocess(context.TODO(), nil, nil)
+	if err == nil {
+		t.Fatalf("not working yet")
+	}
+}
+
+// fakes
+
+type fakeTRepo struct {
+	template.Repository
+	t map[string]template.Info
+}
+
+func (t *fakeTRepo) GetTemplateInfo(templateName string) (template.Info, bool) {
+	ti, found := t.t[templateName]
+	return ti, found
+}
+
+type fakeRuntimeState struct {
+	ra  []*RuntimeAction
+	err error
+	hs  *HandlerState
+}
+
```

Comment:
fetch is not a concern of the dispatcher.

Those tests will be covered in  RuntimeState tests.